src/main/java/org/fogbowcloud/arrebol/api/http/controllers/JobController.java

```diff
@@ -29,19 +30,29 @@ public JobController(JobService jobService){
     public ResponseEntity<String> addJob(@RequestBody JobSpec jobSpec){
         LOGGER.info("Adding new Job: " + jobSpec + ".");
 
-        String jobId = jobService.addJob(jobSpec);
-        JobResponse jobResponse = new JobResponse(jobId);
+        try{
+            String jobId = jobService.addJob(jobSpec);
+            JobResponse jobResponse = new JobResponse(jobId);
 
-        LOGGER.info("Added " + jobSpec.getLabel() + " with id " + jobId + ".");
-        return new ResponseEntity(jobResponse, HttpStatus.CREATED);
+            LOGGER.info("Added " + jobSpec.getLabel() + " with id " + jobId + ".");
+            return new ResponseEntity(jobResponse, HttpStatus.CREATED);
+        } catch (Exception e){
+            LOGGER.error(String.format(Messages.Exception.GENERIC_EXCEPTION, e.getMessage()), e);
+            throw e;
+        }
     }
 
     @RequestMapping(value = ApiDocumentation.ApiEndpoints.JOB_PATH, method = RequestMethod.GET)
     public ResponseEntity<String> getJob(@PathVariable String id){
         LOGGER.info("Getting an job with id: " + id);
 
-        Job job = jobService.getJobById(id);
-        return new ResponseEntity(job, HttpStatus.CREATED);
+        try {
+            Job job = jobService.getJobById(id);
+            return new ResponseEntity(job, HttpStatus.CREATED);
+        } catch(Exception e){
+            LOGGER.error(String.format(Messages.Exception.GENERIC_EXCEPTION, e.getMessage()), e);
```

Comment:
I think we need to add the id here in the log. In this way, it is easy to debug when shit happens

Reply:
Done.